### PR TITLE
Use Annotations for common knative validation.

### PR DESCRIPTION
**Pull request by n3wscott**
Here is the start of me looking at replacing some of the validation we do with annotations.

Please let me know if you like this direction of if I should stop.

### CODE REVIEW

**Comment by n3wscott** [comment]
Alternates: https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/#advanced-topics Heading: Validation

We could use OpenAPI to do validation. But I think the downside is it is all or nothing with OpenAPI.

**Comment by n3wscott** [comment]
/hold

@n3wscott to make a PR with the same functionality but as an OpenAPI config.

**Comment by n3wscott** [comment]
/hold cancel

I think this is still a good idea... 

/assign @mattmoor

File: apis/validators/k8s_qualified_name.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+func (v *K8sQualifiedNameValidator) OnParent() bool {
+	return false
+}
+
+func (v *K8sQualifiedNameValidator) OnField() bool {
+	return true
+}
+
+func (v *K8sQualifiedNameValidator) Validate(fieldName string, value interface{}) (bool, *apis.FieldError) {
```
Comment (ImJasonH) [inline apis/validators/k8s_qualified_name.go]: Could `Validate` just return err, and absence of error means it's valid?

**Comment by n3wscott** [comment]
I am gonna close this now. Learning about annotations in go was interesting but the inability to annotate internal fields from refed objects makes this not useful.

**Pull request closed without merging**